Classes/common/CDTIncrementalStore.m

```diff
@@ -0,0 +1,2798 @@
+            NSString *bytes =
+                [self encodeBlob:data withName:name inStore:blobStore withMIMEType:mimeType];
+            return @{
+                name : bytes,
+                MakeMeta(name) : @{kCDTISMIMETypeKey : @"application/octet-stream"}
+            };
+        }
+        case NSTransformableAttributeType: {
+            NSString *xname = [attribute valueTransformerName];
+            NSString *mimeType = @"application/octet-stream";
+            Class myClass = NSClassFromString(xname);
+            // Yes, we could try/catch here.. but why?
+            if ([myClass respondsToSelector:@selector(MIMEType)]) {
+                mimeType = [myClass performSelector:@selector(MIMEType)];
+            }
+            id xform = [[myClass alloc] init];
+            // use reverseTransformedValue to come back
+            NSData *save = [xform transformedValue:obj];
+            NSString *bytes =
+                [self encodeBlob:save withName:name inStore:blobStore withMIMEType:mimeType];
+
+            return @{ name : bytes, MakeMeta(name) : @{kCDTISMIMETypeKey : mimeType} };
+        }
+        case NSObjectIDAttributeType: {
+            // I don't think converting to a ref is needed, besides we
+            // would need the entity id to decode.
+            NSManagedObjectID *oid = obj;
+            NSURL *uri = [oid URIRepresentation];
+            return @{
+                name : [uri absoluteString],
+            };
+        }
+        case NSDecimalAttributeType: {
+            NSDecimalNumber *dec = obj;
+            NSString *desc = [dec description];
+            NSDecimal val = [dec decimalValue];
+            NSData *data = [NSData dataWithBytes:&val length:sizeof(val)];
+            NSString *b64 = [data base64EncodedStringWithOptions:0];
+            NSMutableDictionary *meta = [NSMutableDictionary dictionary];
+            meta[kCDTISDecimalImageKey] = b64;
+
+            if ([dec isEqual:[NSDecimalNumber notANumber]]) {
+                meta[kCDTISFPNaNKey] = @"true";
+                desc = @"0";
+            }
+
+            return @{name : desc, MakeMeta(name) : [NSDictionary dictionaryWithDictionary:meta]};
+        }
+        case NSDoubleAttributeType: {
+            NSNumber *num = obj;
+            double dbl = [num doubleValue];
+            NSNumber *i64 = @(*(int64_t *)&dbl);
+            NSMutableDictionary *meta = [NSMutableDictionary dictionary];
+            meta[kCDTISDoubleImageKey] = i64;
+
+            if ([num isEqual:@(INFINITY)]) {
+                num = @(DBL_MAX);
+                meta[kCDTISFPInfinityKey] = @"true";
+            }
+            if ([num isEqual:@(-INFINITY)]) {
+                num = @(-DBL_MAX);
+                meta[kCDTISFPNegInfinityKey] = @"true";
+            }
+            if ([num isEqual:@(NAN)]) {
+                num = @(0);  // not sure what to do here
+                dbl = 0.;
+                meta[kCDTISFPNaNKey] = @"true";
+            }
```

Comment:
There is a brief discussion on this [here](https://github.com/jimix/CDTDatastore/pull/18/files#diff-a19a528c6ca4f3a8a6d02158a1eaf4d8R498)
So there are really three goals here:

1. provide a JSON compatible `Number` for `NaN` et. al. that can be useful for expressions
  - using `0` for `NaN` might be dubious here, but I can't think of a better value
1. detect `NaN` et. al. without needing to know IEEE-754
1. use IEEE-754 as an absolute store

Having said that, two things I've discovered recently:

1. using `Booleans` with three object keys is cumbersome, better to have a string object with three possible values.
 - I will likely change this soon.
1. If a non-IEEE-754 capable client updates the JSON value, how can it update the image?
  - don't have anything in mind for this